qcodes/tests/dataset/test_dataset_export.py

```diff
@@ -464,6 +492,31 @@ def test_to_xarray_da_dict_paramspec_metadata_is_preserved(mock_dataset_label_un
             assert xr_da.attrs[spec_name] == spec_value
 
 
+def test_inverted_coords_perserved_on_netcdf_roundtrip(
+    tmp_path_factory, mock_dataset_inverted_coords
+):
+    tmp_path = tmp_path_factory.mktemp("export_netcdf")
+    path = str(tmp_path)
+
+    xr_ds = mock_dataset_inverted_coords.to_xarray_dataset()
+    assert xr_ds["z1"].dims == ("x", "y")
+    assert xr_ds["z2"].dims == ("y", "x")
+    mock_dataset_inverted_coords.export(
+        export_type="netcdf", path=tmp_path, prefix="qcodes_"
+    )
+
+    expected_path = f"qcodes_{mock_dataset_inverted_coords.captured_run_id}_{mock_dataset_inverted_coords.guid}.nc"
+    assert os.listdir(path) == [expected_path]
+    file_path = os.path.join(path, expected_path)
+    ds = qcodes.dataset.load_from_netcdf(file_path)
+
+    with pytest.warns(UserWarning):
+        xr_ds_reimported = ds.to_xarray_dataset()
+    assert xr_ds_reimported["z1"].dims == ("x", "y")
+    assert xr_ds_reimported["z2"].dims == ("y", "x")
+    assert xr_ds == xr_ds_reimported
```

Comment:
perhaps use identical for strictness https://docs.xarray.dev/en/stable/user-guide/combining.html?highlight=equality#equals-and-identical ? or even https://docs.xarray.dev/en/stable/generated/[xarray.testing.assert_identical](https://docs.xarray.dev/en/stable/generated/xarray.testing.assert_identical.html#xarray-testing-assert-identical).html#xarray-testing-assert-identical
```suggestion
    assert xr_ds.identical(xr_ds_imported)
```

Reply:
done, required reordering so that conversion is done after export (so both have the the same export info attr)